docs/fsharp/style-guide/conventions.md

```diff
@@ -111,10 +113,148 @@ This enables the following:
 
 One could even argue that the example with classes is more "functional" than the example with let-bound values in a module.
 
-## Prefer error handling with exceptions in most cases
+## Error handling
+
+Error handling in large systems is a difficult endeavor, and there are no silver bullets in ensuring your systems are fault-tolerant and behave well. The following guidelines should help in this difficult space.
+
+### Represent error cases and illegal state in your types
+
+With Discriminated Unions, F# gives you the ability to represent faulty program state in your type system. For example:
+
+```fsharp
+type MoneyWithdrawalResult =
+    | Success of amount:decimal
+    | InsufficientFunds of balance:decimal
+    | CardExpired of DateTime
+    | UndisclosedFailure
+```
+
+In this case, there are three known ways that withdrawing money from a bank account can fail. Each error case is represented in the type, and can thus be dealt with safely throughout the program.
+
+```fsharp
+let handleWithdrawal amount =
+    let w = withdrawMoney amount
+    match w with
+    | Success am -> printfn "Successfully withdrew %f" am
+    | InsufficientFunds balance -> printfn "Failed: balance is %f" balance
+    | CardExpired expiredDate -> printfn "Failed: card expired on %O" expiredDate
+    | UndisclosedFailure -> printfn "Failed: unknown - retry perhaps?"
+```
+
+In general, if you can model the different ways that something can **fail** in your domain, then error handling code is no longer treated as something you must deal with in addition to regular program flow. It is simply a part of normal program flow. This also makes it easier to maintain as your domain changes over time.
```

Comment:
might be worth mentioning that explicit error cases are more amenable to unit testing.